packages/puppeteer-core/src/common/ElementHandle.ts

```diff
@@ -65,6 +66,7 @@ export class CDPElementHandle<
     return this.handle.client;
   }
 
+  @throwIfDisposed()
   override remoteObject(): Protocol.Runtime.RemoteObject {
```

Comment:
I think this might not need to throw. It does not make any calls and you might still want to have the remote object for the disposed handle.

Reply:
But is the remote object still available after it has been disposed?

Reply:
remote object is the metadata about the object, not the object itself.

Reply:
by looking at the code I don't think the metadata is deleted and I am not sure if it necessarily should be.

Reply:
The metadata isn't deleted, but it contains information about the handle such as the objectId which is no longer valid after the handle is disposed.

Reply:
<img width="1397" alt="Screenshot 2023-08-31 at 13 29 40" src="https://github.com/puppeteer/puppeteer/assets/399150/e1c7bbe5-84cc-4482-88ca-5ac696cf486b">

the rest is valid though. E.g., the serialised value of the disposed object.